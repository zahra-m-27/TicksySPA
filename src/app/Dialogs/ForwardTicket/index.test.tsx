import ReactDOM from 'react-dom';
import React, {useRef} from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ForwardTicketDialog from './index';

const Sample: React.FC = ({}) => {
  const ref = useRef(() => {});
  return <ForwardTicketDialog onDismissRef={ref} />;
};

const sample = <Sample />;

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('render header', () => {
  const {getByText} = render(sample);
  getByText(/From:/i);
});

it('render topic label', () => {
  const {getByText} = render(sample);
  getByText(/:عنوان تاپیک/i);
});

it('render buttons', () => {
  const {getByText} = render(sample);
  getByText(/ارسال/i);
  getByText(/انصراف/i);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
