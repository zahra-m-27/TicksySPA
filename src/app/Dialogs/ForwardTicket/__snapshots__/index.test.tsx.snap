// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <img
      className="arrow_right"
      src="arrow_right.svg"
    />
    <img
      className="arrow_down"
      src="arrow_down.png"
    />
    <div
      className="spliter"
    />
    <span
      className="from"
    >
      From:
    </span>
    <span>
       سید علی علوی مسئول خدمات آموزش مجازی
    </span>
    <div
      className="close_container"
    >
      <img
        className="close"
        data-testid="close-button"
        onClick={[Function]}
        src="close_2.svg"
      />
    </div>
  </div>
  <div
    className="content"
  >
    <span
      className="label"
    >
      :عنوان تاپیک
    </span>
    <div
      className="container topic_drop_down"
      data-open={false}
    >
      <p
        className="label"
      />
      <div
        className="content"
        data-testid="dropdown-container"
        onClick={[Function]}
      >
        <img
          alt="arrow down"
          className="arrow_down"
          src="arrow_down.svg"
        />
        <p
          className="selected_item_label"
        >
          انتخاب كنيد
        </p>
      </div>
      <div
        className="items"
      >
        <p
          className="item"
          onClick={[Function]}
        >
          تست 1
        </p>
        <p
          className="item"
          onClick={[Function]}
        >
          تست 2
        </p>
      </div>
    </div>
    <span
      className="label"
    >
      :دسته بندی
    </span>
    <div
      className="no_topic"
    >
      <img
        src="papers.png"
      />
      <p
        dir="auto"
      >
        !شما هنوز تاپیکی را انتخاب نکرده اید
      </p>
    </div>
  </div>
  <div
    className="footer"
  >
    <div
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#0088e3",
        }
      }
    >
      <span>
        ارسال
      </span>
    </div>
    <div
      className="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#606f7b",
        }
      }
    >
      <span>
        انصراف
      </span>
    </div>
  </div>
</div>
`;
